data_manager_select_index_by_path/data_manager/path_name_value_key_manager.py

```diff
@@ -1,104 +1,134 @@
+        self.indexes_properties_file = indexes_properties_file
+
+        self.check_params()
+
+        self.index_properties = self.get_index_properties()
+
+        self.check_index_file_presence()
+
+    def check_params(self):
+
+        check_tab('name', self.name)
+        check_tab('index_path', self.index_path.absolute().__str__())
+        check_tab('value', self.value)
+        check_tab('dbkey', self.dbkey)
+
+    def get_index_properties(self) -> dict:
+        with self.indexes_properties_file.open('r') as properties_file:
+            indexes = yaml.safe_load(properties_file)
+        index_properties = indexes.get(self.data_table_name)
+        if not index_properties:
+            raise ValueError("{0} not a supported table name".format(self.data_table_name))
+        return index_properties
+
+    def check_index_file_presence(self):
+        index_name = self.index_properties.get('name')
+        if not index_name:
+            raise ValueError(
+                "Property 'name' not defined for '{0}', please contact the developers to correct the mistake.")
+        index_extensions = self.index_properties.get('extensions', [''])
+
+        # Sometimes an index path is a prefix.
+        # For example, with BWA. 'reference.fa' is the index. But the actual index files are
+        # 'reference.fa.amb', 'reference.fa.ann' etc.
+
+        # If the index is not a prefix, the index file is taken to be the path itself.
+        index_is_a_prefix = self.index_properties.get('prefix', True)
+        if index_is_a_prefix:
+            for extension in index_extensions:
+                if not prefix_plus_extension_exists(self.index_path.parent, self.index_path.name, extension):
+                    raise FileNotFoundError(
+                        'Unable to find files with prefix "{0}" and extension "{1}" in {2}. Is this a valid {3}?'
+                        .format(self.index_path.name, extension, self.index_path.parent, index_name))
+        else:
+            if not self.index_path.exists():
+                raise FileNotFoundError('Unable to find path {0}.'.format(self.index_path))
+
+    @property
+    def data_manager_dict(self) -> dict:
+        data_table_entry = dict(value=self.value, dbkey=self.dbkey, name=self.name, path=self.index_path.__str__())
+        data_manager_dict = dict()
+        data_manager_dict[self.data_table_name] = data_table_entry
+        return data_manager_dict
+
+    @property
+    def data_manager_json(self) -> str:
+        return json.dumps(self.data_manager_dict)
+
+
 def main():
+    options = argument_parser().parse_args()
 
-    #value = "test_value"
-    #name = "test_name"
-    #print '{0} other {1} more{0}'.format(value, name )
-    #print '{0} is not a valid {1}. It may not contain a tab.'.format( value, name )
+    if options.json_output_file.exists():
+        raise FileExistsError("'{0}' already exists.".format(str(options.json_output_file)))
```

Comment:
Why not just overwrite the existing file?

Reply:
This is a galaxy tool. This file should not exist, otherwise something is seriously wrong.